### PR TITLE
ColumnPairTrends score depends on the data index

### LINKED ISSUE #582: ColumnPairTrends score depends on the data index
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDMetrics version: 0.14.1

### Error Description

By design, the metrics and property score should be independent of the indexes of the real and synthetic data.
However, this is currently not the case for the ColumnPairTrends property as shown below. The issue comes from the discretization step when `numerical` and `datetime` columns are converted to `categorical`, the indexes are not preserved.

### Steps to reproduce

The code below should output a metric score of 1.0 since real and synthetic data are the same (the only have different indexes).

```python
import pandas as pd
from sdmetrics.reports.single_table._properties import ColumnPairTrends

real_data = pd.DataFrame({
    'A': [1, 2, 3],
    'B': ['a', 'b', 'a']
}, index=[0, 1, 2])

synthetic_data = pd.DataFrame({
    'A': [1, 2, 3],
    'B': ['a', 'b', 'a']
}, index=[0, 4, 2])

metadata = {
    'columns': {
        'A': {'sdtype': 'numerical'},
        'B': {'sdtype': 'categorical'}
    }
}

property = ColumnPairTrends()
property._generate_details(real_data, synthetic_data, metadata)
```
The current output is:

Screenshot 2024-06-07 at 10 42 22

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86b0tvv64
Resolve #582

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0tvv64 SDMetrics - ColumnPairTrends property depends on the data index #582](https://app.clickup.com/t/86b0tvv64)

File: sdmetrics/reports/single_table/_properties/column_pair_trends.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def _get_processed_data(self, data, metadata):
                     column_name, processed_data[column_name]
                 )
 
-        return processed_data, pd.DataFrame(discretized_dict)
+        return processed_data, pd.DataFrame(discretized_dict, index=processed_data.index)
```
Comment (lajohn4747) [inline sdmetrics/reports/single_table/_properties/column_pair_trends.py:126]: Just checking, is there always an `.index` provided?

Comment (R-Palazzo) [reply sdmetrics/reports/single_table/_properties/column_pair_trends.py:126]: The method expects a pandas dataframe as input, so I guess there will always be a `.index` if this is the question

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Review by lajohn4747 — APPROVED** [review]

**Pull request merged**